CMakeLists.txt

```diff
@@ -135,7 +135,7 @@ if(NOT TARGET CURL::CURL)
       #INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}"
       INTERFACE_INCLUDE_DIRECTORIES "${CURL_INCLUDE_DIR}"
       INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
-    if(NOT WIN32)
+    if(NOT MSVC)
```

Comment:
I'm curious, what is the MinGW issue with cURL here?

Reply:
just rebuilding to paste the error. it was some linking issues. don't get me wrong, I'm way out of my league with cmake stuff. However, replacing with WIN32 worked;)

Reply:
```
[ 89%] Linking CXX shared library libvalhalla.dll
cd /workspace/valhalla_previous/build/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/valhalla.dir/link.txt --verbose=1
/usr/bin/cmake -E rm -f CMakeFiles/valhalla.dir/objects.a
/usr/bin/x86_64-w64-mingw32-ar cr CMakeFiles/valhalla.dir/objects.a @CMakeFiles/valhalla.dir/objects1.rsp
/usr/bin/x86_64-w64-mingw32-g++  -fdiagnostics-color=auto  -O2 -DNDEBUG -fstack-protector -lssp  -shared -o libvalhalla.dll -Wl,--out-implib,libvalhalla.dll.a -Wl,--major-image-version,3,--minor-image-version,0 -Wl,--whole-archive CMakeFiles/valhalla.dir/objects.a -Wl,--no-whole-archive @CMakeFiles/valhalla.dir/linklibs.rsp
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text+0x35): undefined reference to `curl_global_cleanup'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text+0x79): undefined reference to `curl_easy_cleanup'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0xe2): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x19c): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x254): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x314): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x3cf): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x467): more undefined references to `curl_easy_setopt' follow
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x5ae): undefined reference to `curl_easy_perform'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x5e3): undefined reference to `curl_easy_getinfo'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE[_ZNK8valhalla5baldr8curler_t7pimpl_t5fetchERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERlbPKSt8functionIFvvEE]+0x682): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x39): undefined reference to `curl_easy_init'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x14c): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x210): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x2c2): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x388): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x44e): undefined reference to `curl_easy_setopt'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x49e): undefined reference to `curl_global_init'
/usr/lib/gcc/x86_64-w64-mingw32/10.2.1/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/valhalla.dir/objects.a(curler.cc.obj):curler.cc:(.text$_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN8valhalla5baldr8curler_t7pimpl_tC1ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x6d6): undefined reference to `curl_easy_cleanup'
collect2: error: ld returned 1 exit status
```

Reply:
No worries, we're all learning here :)

It looks like cURL library is not found by CMake, but then this `find_package` here should fail as cURL is required
https://github.com/valhalla/valhalla/blob/bf1763c3972dfc7f8ab42bc0accf5c536995b457/CMakeLists.txt#L131-L133

Reply:
Right.. It does find curl while configuring and doesn't while compiling. Specifying `-DCURL_LIBRARY` or `-DCURL_STATICLIB=ON` while configuring doesn't work either. 

If I understand correctly it's now using (or trying to use) the DLL instead of static, right? Is that a problem? When using dynamic lib the compiler does check the symbols exist in the DLL right? I didn't try to run anything which needs cURL, so at this point no idea if runtime is fine.

Reply:
No matter if `MSVC` or `WIN32` in `CMakeLists.txt` CMake always finds the static library:
`-- Found CURL: /usr/x86_64-w64-mingw32/sys-root/mingw/lib/libcurl.dll.a (found version "7.71.1")`

Reply:
> If I understand correctly it's now using (or trying to use) the DLL instead of static, right? Is that a problem? 

No, it's not.
I mean, if both, MinGW and MSVC, can compile and link with HTTP support fine, then it's good.